### ISSUE TITLE
Group owners and privacy settings

### ISSUE THREAD

**Issue by nafens** [iionly/private_profiles#7]
Hi, 

I think I have found an issue with this -very useful- plugin and group owners.

Scenario:
A group owner who owns the group named "**A very public group**" has answered the question:
Who should be able to see your public activity and membership listing?
To:
**Logged-in members only**

Result:
The group "A very public group" is not listed in the list of site groups
https://mysite/groups
even though the group access settings are set to "public", ie its settings are:
Who can see this group? : **Everyone**
Accessibility of group content: **Unrestricted**
Group default access: **Public**

Users that are not logged in and try to go to the group profile page
https://mysite/groups/profile/2222222/a-very-public-group
are redirected to the login page.

Best,

**Comment by nafens** [comment]
I forgot to add the environment where the above scenario takes place

Plugin: Private Profiles 2.3.0
Elgg version:  2.3.14

**Comment by iionly** [comment]
Problem here is the creation of a group is also a user activity that the user decided to hide from "public" (visitors not logged-in). Either way (blocking access or not blocking) seems to cause an inconsistency / problem.

I have to admit that I wasn't aware that this option would block access to group content if the group owner (not necessarily the group members) wants the activity hidden. I don't make use of this option myself to be honest. I can try to look into it when I have some time to spare (I'm afraid that won't happen anytime soon) but I can't promise to alter the current behavior because any additional exceptions from the rules could result in too much load (maybe it turns out this is not the case but I can't say at the moment).

**Comment by nafens** [comment]
Thanks for answering. 

Also it was strange for us when a group owner complained that his group was not visible, and at first we were sure that he had messed up his group access settings. When we ensured that access settings were set to public  it took us a while to conclude that the problem was related to his private profile settings as we never thought that this plugin had  any effect on group visibility.

I believe a workaround would be to implement a check if the logged in user is the owner any group and if he is,  then  disable/hide the "Who should be able to see your public activity and membership listing?"  setting in the plugin form. 
Alternatively don't hide the setting but no matter what group owners answer in this question, always make this setting answered as "everyone" for group owners. 

Best,

**Comment by iionly** [comment]
I don't think that either of your suggestions will work. Hiding the option might result sooner or later in a user who has created a group asking why this option is not available for him. And it will also not work with suddenly hiding this option on an existing site after upgrading the plugin when a user has already activated this option. It's not possible for the same reason you can not simply ignore the selection and always display their content for everyone. That's the whole point of this option: to NOT show their activity / content to non-members. Ignoring this wish would be an extreme violation of trust and privacy.

Strictly speaking, the option works exactly as it's supposed to work: it does hide the user's activity and this activity also includes creation of content and creation of groups. It's maybe first of all a matter of explaing this better in the option description.

As I already mentioned in my last reply, I see the only possible solution in adding a second option that would explicitely deal with visibility of groups. But this option would only make sense if it wouldn't slow down the query too much. Whenever I have time (not in near future) I can try it.
